OCPBUGS-4279: Upgrade pf 2022 13

jgbernalp opened the pull request:
Fixes: https://issues.redhat.com/browse/OCPBUGS-4279

Based on https://github.com/openshift/console/pull/12146

jgbernalp commented:
https://github.com/openshift/console/pull/12330 update is still needed as the main issue is the TS version we are using in the console. From `victory-core@36.5.0` TS is used directly instead of declaration files, and the declaration merging [here](https://github.com/FormidableLabs/victory/blob/main/packages/victory-core/src/victory-portal/victory-portal.tsx#L13) does not play well with the console TS version, leading to the issue. 

Should we consider upgrading TS version as well?

cc @jhadvig @jcaianirh

jcaianirh reviewed (changes requested):
@jgbernalp we shouldn't be adding victory-core as a dependency to the console. The package puts us over our bundle size limit, and we get this dependency through patternfly. The actual bug is to not create or use custom victory - core components in the console and use patternfly components instead. Unless the code in monitoring is compatible with the version of victory that pf release thats another option but that also keeps us having to update custom code every time pf updates their version of victory-core

jcaianirh commented:
> #12330 update is still needed as the main issue is the TS version we are using in the console. Since `victory-core@36.5.0` TS is used directly instead of declaration files, and the declaration merging [here](https://github.com/FormidableLabs/victory/blob/main/packages/victory-core/src/victory-portal/victory-portal.tsx#L13) does not play well with the console TS version, leading to the issue.
> 
> Should we consider upgrading TS version as well?
> 
> cc @jhadvig @jcaianirh

yes you're right, we do also need to update the console ts version. cc @jhadvig

kyoto commented on frontend/public/components/monitoring/query-browser.tsx:
```diff
@@ -70,6 +70,8 @@ import { PrometheusAPIError } from './types';
 const spans = ['5m', '15m', '30m', '1h', '2h', '6h', '12h', '1d', '2d', '1w', '2w'];
 export const colors = queryBrowserTheme.line.colorScale;
 
+const SafeVictoryPortal = (VictoryPortal as unknown) as React.ComponentType;
```
@jgbernalp Could you please add a code comment explaining what this is for?

jgbernalp replied:
@kyoto Sorry, forgot to put the WIP. I wanted to run the test to track the root issue.

kyoto replied:
👍

jgbernalp commented:
@jcaianirh You are right, adding the `victory-core` dependency just adds 2 versions into the main bundle. Digging deeper onto the issue I found the problem was caused by the way the victory-core imports React, assuming `allowSyntheticDefaultImports`.  I added this option to our `tsconfig` and fixed some other type issues generated by the patterfly version upgrade. Please check 👀.

We are creating custom components in the metrics because of performance issues when loading several series inside the tooltip, as part of our migration to a monitoring dynamic plugin we are currently refactoring components to use native patternfly components and we'll look into this tooltip component as a follow up [task](https://issues.redhat.com/browse/OU-127). 

Side note: While checking the problem I found out that our `@types/react` and `@types/react-dom` (`16.8.13`) differ from our react version (`17.0.1`). Should we create an issue for this along the TypeScript version upgrade?

cc @kyoto @jhadvig

jcaianirh commented:
/jira refresh

jcaianirh commented:
Failure:
[36mINFO[0m[2022-12-06T20:19:35Z] Running step e2e-gcp-console-gather-extra.   
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:164","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2022-12-06T20:23:16Z"}
[36mINFO[0m[2022-12-06T20:23:16Z] Received signal.                              [36msignal[0m=interrupt
[36mINFO[0m[2022-12-06T20:23:18Z] error: Process interrupted with signal interrupt, cancelling execution... 
[36mINFO[0m[2022-12-06T20:23:18Z] cleanup: Deleting release pod release-latest 
[36mINFO[0m[2022-12-06T20:24:35Z] Step e2e-gcp-console-gather-extra succeeded after 5m0s. 
[36mINFO[0m[2022-12-06T20:24:35Z] Running step e2e-gcp-console-gather-audit-logs. 
[36mINFO[0m[2022-12-06T20:26:25Z] Step e2e-gcp-console-gather-audit-logs succeeded after 1m50s. 
[36mINFO[0m[2022-12-06T20:26:25Z] Running step e2e-gcp-console-ipi-deprovision-deprovision. 
[36mINFO[0m[2022-12-06T20:33:15Z] Step e2e-gcp-console-ipi-deprovision-deprovision succeeded after 6m50s. 
[36mINFO[0m[2022-12-06T20:33:15Z] Step phase post succeeded after 22m20s.      
[36mINFO[0m[2022-12-06T20:33:15Z] Releasing leases for test e2e-gcp-console    
[36mINFO[0m[2022-12-06T20:33:15Z] Ran for 4h9m58s                              
[31mERRO[0m[2022-12-06T20:33:15Z] Some steps failed:                           
[31mERRO[0m[2022-12-06T20:33:15Z] 
  * could not run steps: execution cancelled 
[36mINFO[0m[2022-12-06T20:33:15Z] Reporting job state 'failed' with reason 'executing_graph:interrupted' 
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:251","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process gracefully exited before 1h0m0s grace period","severity":"error","time":"2022-12-06T20:33:15Z"}

jcaianirh commented:
/retest

jcaianirh commented:
/retest

jgbernalp commented:
/retest

jcaianirh commented:
/retest

jgbernalp commented:
/retest

jgbernalp commented:
/retest

jcaianirh commented:
/retest

jcaianirh commented:
/retest

juzhao commented:
/label qe-approved

rhamilto commented:
/approve
/lgtm
